rpg_game.py

```diff
@@ -54,29 +114,30 @@ def __init__(self):
         self.coins = 10
         self.armor = 0
         self.evade = 0
-    
+
+
     def attack(self, enemy):
         if random.random() <= 0.2:
-            enemy.health -= self.power*2
+            enemy.health -= self.power * 2
         elif self.armor > 0:
             arm_health = self.health + self.armor
             arm_health -= enemy.power
             print('Your armor has protected you!')
         super(Hero, self).attack(enemy)
 
-        
+
     def buy(self, item):
         self.coins -= item.cost
-        item.apply(hero)
-    
+        item.apply(self)
```

Comment:
I'm by no means a master of OOP, but this is something I saw a lot throughout the script. You'll want to make sure the apply the item to whoever is buying it. As you only have one hero in the game currently this isn't an issue, but if you had multiple heros, or wanted to give monsters the ability to apply items to themselves this could be troublesome.

Reply:
You are absolutely correct. If I had more than one character who could apply items to themselves I wouldn't have put item.apply(hero).